SimG4CMS/Calo/src/HGCSD.cc

```diff
@@ -148,43 +148,37 @@ uint32_t HGCSD::setDetUnitId(const G4Step * aStep) {
   ForwardSubdetector subdet = myFwdSubdet_;
 
   int layer(0), module(0), cell(0);
-  if (m_mode == HGCalGeometryMode::Square) {
+  if (touch->GetHistoryDepth() == levelT_) {
     layer  = touch->GetReplicaNumber(0);
-    module = touch->GetReplicaNumber(1);
-  } else {
-    if (touch->GetHistoryDepth() == levelT_) {
-      layer  = touch->GetReplicaNumber(0);
-      module = -1;
-      cell   = -1;
+    module = -1;
+    cell   = -1;
```

Comment:
@bsunanda , why we need -1 here? initially they were initialized as 0.

Reply:
I shall in any case revise HGCSD later when I have the new version of HGCal geometry.  I want to remove only the square cell part right now to ease the changes needed for the future version

Reply:
There is some intricacies for partial layers - I shall see what can be done